doc/source/data/transforming-datasets.rst

```diff
@@ -78,34 +80,338 @@ express your customized business logic in transformations. Here we will focus on
 :meth:`.map_batches() <ray.data.Dataset.map_batches>` as it's the primary mapping
 API in Datasets.
 
-A UDF can be a function or a callable class, which has the following input/output:
+A batch UDF can be a function or, if using the actor compute strategy, a callable class.
+These UDFs have several batch format options, which control the format of the
+batches that are passed to the provided batch UDF. Depending on the underlying dataset
+format, using a particular batch format may or may not incur a data conversion cost
```

Comment:
From my understanding, we can talk about cost later.
I think the first things user will ask are: what's input and output to a UDF? Given that, how do I write a UDF on that input and to produce a desired output?
Once we get them onboarded with that, we can walk them through the cost implications.